features/telemetry/adapters/netflow/src/main/java/org/opennms/netmgt/telemetry/adapters/netflow/Netflow5Adapter.java

```diff
@@ -79,9 +115,23 @@
     // measures the flows/seconds throughput
     private Meter meter;
 
+    // Caches NodeInfo data
+    private LoadingCache<NodeInfoKey, Optional<NodeInfo>> nodeInfoCache;
+
     @PostConstruct
     public void init() {
         meter = metricRegistry.meter("persistence");
+
+        // TODO MVR make this configurable, when it is actually an osgi-module
+        nodeInfoCache = CacheBuilder.newBuilder()
+                .maximumSize(1000)
+                .expireAfterAccess(300, TimeUnit.SECONDS) // 5 Minutes
```

Comment:
I think we should use `expireAfterWrite` instead of `expireAfterAccess` here to ensure that we don't cache these objects for any longer than the configured time - the node details may change.

Reply:
Yep. Makes sense.